controller/aks-cluster-config-handler.go

```diff
@@ -593,6 +593,12 @@ func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.Secr
 		upstreamSpec.WindowsAdminPassword = windowsProfile.AdminPassword
 	}
 
+	// set addons profile
+	addonProfile := clusterState.AddonProfiles
```

Comment:
Would this ever be nil?

Reply:
I have added additional check to verify, when it might be `nil`

Reply:
It's fine to have it, but you don't need it: reading from a `nil` map is valid in Go.